lib/data/kinetic/Kinetic.js

```diff
@@ -0,0 +1,302 @@
+            this._events.emit(
+                'getResponse', null, Buffer.concat(this._chunkTab));
+        }
+    }
+
+    _newSocket() {
+        const socket = new net.Socket({ allowHalfOpen: false }).pause();
+        socket.setKeepAlive(true);
+        socket.unref();
+        return socket;
+    }
+
+    _propError(pdu) {
+        const statusCode = pdu.getStatusCode();
+        if (statusCode === kinetic.errors.NOT_FOUND) {
+            return errors.ObjNotFound;
+        }
+        if (statusCode !== kinetic.errors.SUCCESS) {
+            return pdu.getErrorMessage();
+        }
+        return undefined;
+    }
+
+    getConnectionId() {
+        return this._connectionId;
+    }
+
+    getDrives() {
+        return this._drives;
+    }
+
+    getPDU() {
+        return this._initPDU;
+    }
+
+    getSocket() {
+        return this._socket;
+    }
+
+    setSequence(index, sequence) {
+        if (sequence >= Number.MAX_VALUE) {
+            const temp = this._sockets[index];
+            this._sockets[index] = this._newSocket();
+            this.connect();
+            temp.destroy();
+            this._sockets[index].sequence = 0;
+        } else {
+            this._sockets[index].sequence = sequence;
+        }
+        return this;
+    }
+
+    put(value, options, callback) {
+        this._getDrive((err, socket) => {
+            if (err) {
+                return callback(err);
+            }
+            const key = Buffer.concat(
+                [Buffer.from(`${socket.drive.host}:${socket.drive.port}`),
+                 crypto.randomBytes(6)]);
+            const obj = Buffer.concat(value);
+            const tag = crypto.createHmac('sha1', 'asdfasdf').update(obj)
+                      .digest();
+            const pdu = new kineticlib.PutPDU(
+                socket.sequence, socket.connectionID, socket.clusterVersion,
+                key, obj.length, tag, options);
+            this.setSequence(socket.index, socket.sequence + 1);
+            const header = pdu.read();
+            const len = header.length + obj.length;
+            return socket.sock.write(Buffer.concat([header, obj], len), err => {
+                if (err) {
+                    return callback(err);
+                }
+                return this._events.once(
+                    'putResponse', err => callback(err, key));
+            });
+        });
+    }
+
+    _getDrive(callback) {
+        this._bindGetLog(callback);
+        this._sockets.forEach(socket => {
+            const pdu = new kineticlib.GetLogPDU(
+                socket.sequence,
+                socket.connectionID,
+                socket.clusterVersion,
+                { types: [kineticlib.logs.CAPACITIES] });
+            const header = pdu.read();
+            this.setSequence(socket.index, socket.sequence + 1);
+            socket.sock.write(header, err => {
+                if (err) {
+                    return callback(err);
```

Comment:
It's good for `this._bindGetLog(callback)` but not for `callback(err)` here. If write in multiple sockets fails, multiple callback will return.